tests/src/podman-machine-onboarding.spec.ts

```diff
@@ -0,0 +1,143 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Page } from 'playwright';
+import type { RunnerTestContext } from './testContext/runner-test-context';
+import { afterAll, beforeAll, test, describe, beforeEach } from 'vitest';
+import { PodmanDesktopRunner } from './runner/podman-desktop-runner';
+import { WelcomePage } from './model/pages/welcome-page';
+import type { DashboardPage } from './model/pages/dashboard-page';
+import type { Locator } from '@playwright/test';
+import { expect as playExpect } from '@playwright/test';
+import { NavigationBar } from './model/workbench/navigation';
+import type { SettingsBar } from './model/pages/settings-bar';
+import { ResourcesPage } from './model/pages/resources-page';
+
+const PODMAN_MACHINE_STARTUP_TIMEOUT: number = 60_000;
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let dashboardPage: DashboardPage;
+let resourcesPage: ResourcesPage;
+let settingsBar: SettingsBar;
+let navigationBar: NavigationBar;
+
+let notificationPodmanSetup: Locator;
+let onboardingStepBody: Locator;
+let podmanMachineConfiguration: Locator;
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('podman-extensions-e2e');
+
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  navigationBar = new NavigationBar(page);
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+}, 120000);
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+describe('Podman Machine initialization through Onboarding', async () => {
+  test('Check Podman set-up notification presence (should be visible if machine was not initialized)', async () => {
```

Comment:
Test should be run only if we are expecting push notification. (No need for a change now).

Reply:
The notification check is related to the dashboard notification, not a system one. Should I change the test case description?